diff --git a/node_modules/react-native-iap/android/build.gradle b/node_modules/react-native-iap/android/build.gradle
index 3e7691f..2f5ca6e 100644
--- a/node_modules/react-native-iap/android/build.gradle
+++ b/node_modules/react-native-iap/android/build.gradle
@@ -93,7 +93,7 @@ android {
     externalNativeBuild {
       cmake {
         cppFlags "-frtti -fexceptions -Wall -Wextra -fstack-protector-all"
-        arguments "-DANDROID_STL=c++_shared", "-DANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES=ON"
+        arguments "-DANDROID_STL=c++_shared", "-DANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES=ON", "-DCMAKE_OBJECT_PATH_MAX=1024"
         abiFilters (*reactNativeArchitectures())
 
         buildTypes {
@@ -111,6 +111,7 @@ android {
   externalNativeBuild {
     cmake {
       path "CMakeLists.txt"
+      buildStagingDirectory "C:/cx/iap"
     }
   }
 
